### ISSUE TITLE
YAML configuration file support

### ISSUE THREAD

**Issue by silverwind** [wooorm/remark-lint#75]
YAML is a popular file format for configuration files and is supported by tools like `ESLint` and `Stylelint`, so I think it'd be a great addition to have.

As for an implementation, once could just do `JSON.parse()` and if that fails, fallback to `YAML.parse`.

**Comment by jayvdb** [comment]
IMO TOML would be a better choice that is gaining acceptance/traction.

https://github.com/toml-lang/toml/blob/master/README.md

**Comment by wooorm** [comment]
@silverwind Cool idea! I believe you're suggesting a file named '.remarkrc.yaml', right? that should be doable. Something with js-yaml probably? Would you be interested in working on it? If so, the right place would be 'wooorm/unified-engine', which is externalise from what previously was the remark CLI. If not, I'll see what I can do but it'll take longer :)

@jayvdb that's cool to but I believe it's used a bit less so I'd rather hold off of that for now. Keep in mind that '.remarkrc.js' is also supported, so it would be quite simple to roll it yourself!

**Comment by silverwind** [comment]
Yes, `js-yaml` is my parser of choice. I'm certainly interested in working on it, but I'm not a big fan of `'.remarkrc.yaml` and friends and would prefer a dual-format `.remarkrc`, but it can always be added in a later iteration.

**Comment by silverwind** [comment]
Oh, and has the train in renaming the file to `.remarklintrc` left? Would be more accurate :)

**Comment by wooorm** [comment]
Oh that's another option! We could certainly support a '.remarkrc' file in either JS, JSON, or YAML, but even if we do that I believe we should add support for the extension too; I like the clarity of it.

And about the actual name: as you can also configure remark itself, and other plugins, I'm ffor keeping it at 'remarkrc'!

**Comment by wooorm** [comment]
So, I just added support for `.remarkrc.yaml` files (once released, that is). An example configuration file could look as follows: `.remarkrc.yaml`:

```yaml
output: true
settings:
  commonmark: true
plugins:
  lint:
    no-duplicate-headings: false
```

This can also be passed as `--rc-path` on `remark-cli` (as long as the extension is there).

I opted not add a try/catch-mechanism testing JS, YAML, or JSON, as I kinda dislike the ambiguety of that. However, if you’d like that I’m open to discussion. In that case, could you open an issue at [unified-engine](https://github.com/wooorm/unified-engine)?

Thanks for the suggestion!

**Comment by asbjornu** [comment]
As I much prefer YAML over JSON, many Remark tools such as `remark-cli` doesn't seem to support anything but the JSON format. Shouldn't parsing of config be up to `remark` so every tool doesn't have to add its own configuration format detector and parser?

**Comment by ChristianMurphy** [comment]
@asbjornu https://github.com/remarkjs/remark-lint/issues/75#issuecomment-233926118

**Comment by asbjornu** [comment]
@ChristianMurphy. 🤦 Thanks.
